main.go

```diff
@@ -262,16 +262,16 @@ func authorize(tmpl interface {
 			email = "simulate-delivered@notifications.service.gov.uk"
 		}
 
-		sub := r.FormValue("sub")
-		if sub == "" {
-			if templateSub {
-				sub = randomString("sub-", 20)
-			} else {
-				h := sha256.New()
-				h.Write([]byte(email))
-				encodedEmail := base64.StdEncoding.EncodeToString(h.Sum(nil))
-				sub = "urn:fdc:mock-one-login:2023:" + encodedEmail
-			}
+		h := sha256.New()
+		h.Write([]byte(email))
+		encodedEmail := base64.StdEncoding.EncodeToString(h.Sum(nil))
+		sub := "urn:fdc:mock-one-login:2023:" + encodedEmail
+
+		subject := r.FormValue("subject")
+		if subject == "manual" && r.FormValue("subjectValue") != "" {
```

Comment:
could we have a third option here, so `manual` = existing behaviour of `"sub-" + rand`, `email` = this behaviour, `fixed` = else below?

Reply:
or `manual` = this, `random` = `"sub" + rand`, `fixed` = below. I guess it just depends on how you want to do the UI

Reply:
Ah, yes. I removed that and I see that I misread what it did. I'd assumed it was fundamentally the same as the email generation route but it would be different for every login. I'll rework it with a third option probably as that makes it the most clear as to what it's doing.

Reply:
Are you set on the "sub-" + random format or is something like "urn:fdc:mock-one-login:2023:" + random ok?

Reply:
either format is fine, we don't rely on anything in particular, so may as well go for the more correct version of `"urn:fdc:mock-one-login:2023:" + rand` since you are changing the code